config/validate.go

```diff
@@ -3,24 +3,123 @@ package config
 import (
 	"fmt"
 	"github.com/BurntSushi/toml"
+	"github.com/agext/levenshtein"
+	"reflect"
+	"sort"
+	"strings"
+	"unicode"
 )
 
+// UnexpectedKey represents result of validation
+type UnexpectedKey struct {
+	Name        string
+	SuggestName string
+}
+
+func normalizeKeyname(f reflect.StructField) string {
+	name := string(unicode.ToLower([]rune(f.Name)[0])) + f.Name[1:]
+	if s := f.Tag.Get("toml"); s != "" {
+		name = s
+	}
+	return name
+}
+
+func addKeynameToCandidates(f reflect.StructField, candidates *[]string) *[]string {
+	name := normalizeKeyname(f)
+	*candidates = append(*candidates, name)
+	return candidates
```

Comment:
Are there a reason why this function receives a pointer to slice? In my opinion, a pointer to a slice is rare appearing in the Go code. So, maybe it is sufficient to return a slice:

```go
func addKeynameToCandidates(..., candidate []string) []string {
    ...
    return append(candidates, name)
}
```

and the receiver side:

```go
candidates = addKeynameToCandidates(f, candidates)
```